common/src/main/java/com/microsoft/identity/common/adal/internal/cache/StorageHelper.java

```diff
@@ -225,76 +245,133 @@ public String encrypt(final String clearText)
 
         final String encryptedText = new String(Base64.encode(blobVerAndEncryptedDataAndIVAndMacDigest,
                 Base64.NO_WRAP), AuthenticationConstants.ENCODING_UTF8);
-        Log.v(TAG, "Finished encryption");
+        Logger.verbose(TAG + methodName, "Finished encryption");
 
         return getEncodeVersionLengthPrefix() + ENCODE_VERSION + encryptedText;
     }
 
     @Override
     public String decrypt(final String encryptedBlob) throws GeneralSecurityException, IOException {
         final String methodName = "decrypt";
-        Log.v(TAG, "Starting decryption");
+        Logger.verbose(TAG + methodName, "Starting decryption");
 
         if (StringExtensions.isNullOrBlank(encryptedBlob)) {
             throw new IllegalArgumentException("Input is empty or null");
         }
 
-        int encodeVersionLength = encryptedBlob.charAt(0) - 'a';
-        validateEncodeVersion(encryptedBlob, encodeVersionLength);
+        if (getEncryptionType(encryptedBlob) == EncryptionType.UNENCRYPTED) {
+            Logger.verbose(TAG + methodName, "This string is not encrypted. Finished decryption.");
```

Comment:
This behavior change is introduced because we'll be treating PRT, BRT, etc as encrypted value after this change is shipped.

These values are currently unencrypted, and if the migration fails, then these values will remain unencrypted. 

Technically, we could keep the same behavior, and it would throw an exception (since it fails to verify what type of 'encryption' was used on the blob. We're proposing that if such things happen, then we should fail the flow and wipe the account. 

I think this is a better way to gracefully support such case. However, if you think it's an edge case and not worth making the change, I could revert back to the original logic.